src/confidential/mod.rs

```diff
@@ -0,0 +1,226 @@
+// Miniscript
+// Written in 2022 by
+//     Andrew Poelstra <apoelstra@wpsoftware.net>
+//
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! Confidential Descriptors
+//!
+//! Implements ELIP ????, described at `URL`
+//!
+
+pub mod hash_to_sk;
+pub mod slip77;
+
+use crate::{Error, MiniscriptKey, ToPublicKey};
+use elements::hashes::hex;
+use elements::secp256k1_zkp;
+use crate::extensions::{CovenantExt, CovExtArgs, Extension, ParseableExt};
+use std::{fmt, iter};
+
+/// A description of a blinding key
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+pub enum Key<Pk: MiniscriptKey> {
+    /// Blinding key is computed from the hash of a sorted list of public keys
+    ///
+    /// The first public key is included outside of the `Vec` to reduce allocations
+    /// and to enforce that the list of keys is not empty; but it is not special in
+    /// any way.
+    HashToSk(Pk, Vec<Pk>),
```

Comment:
This makes creating the `Key` really annoying. Can we just have a  Vec<Pk> instead? To ensure that there is at least one element, we should wrap this in a struct and add constructors.

Reply:
IMO, we should really make things convenient for users by adding some extra code and allocations here.

Reply:
lol, okay, agreed